server/svix-server/src/redis/mod.rs

```diff
@@ -1,88 +1,177 @@
 mod cluster;
+mod sentinel;
 
-use std::time::Duration;
+use std::{sync::Arc, time::Duration};
 
 use bb8::{Pool, RunError};
 use bb8_redis::RedisConnectionManager;
-use redis::{FromRedisValue, RedisError, RedisResult};
+use redis::{
+    aio::ConnectionManagerConfig, sentinel::SentinelNodeConnectionInfo, AsyncConnectionConfig,
+    FromRedisValue, ProtocolVersion, RedisConnectionInfo, RedisError, RedisResult, TlsMode,
+};
+use sentinel::RedisSentinelConnectionManager;
+use tokio::sync::Mutex;
 
 pub use self::cluster::RedisClusterConnectionManager;
-use crate::cfg::{CacheBackend, QueueBackend};
+use crate::cfg::{CacheBackend, QueueBackend, SentinelConfig};
 
 pub const REDIS_CONN_TIMEOUT: Duration = Duration::from_secs(2);
 
+pub enum RedisVariant<'a> {
+    Clustered,
+    NonClustered,
+    Sentinel(&'a SentinelConfig),
+}
+
 #[derive(Clone, Debug)]
 pub enum RedisManager {
     Clustered(ClusteredRedisPool),
     ClusteredUnpooled(ClusteredRedisUnpooled),
     NonClustered(NonClusteredRedisPool),
     NonClusteredUnpooled(NonClusteredRedisUnpooled),
+    Sentinel(SentinelPooled),
+    SentinelUnpooled(SentinelUnpooled),
```

Comment:
Not sure the implications, but I wonder how feasible it'd be to split pooled/unpooled out as a different facet entirely so we don't have to have 2 variants for each of these.

This would probably be an upheaval, but can "use a pooler" just be an option on each of these?

Reply:
I have a PR forthcoming that will further refactor the this module. I'll consider this as well.